src/components/AdminDashboard/container.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { PropTypes as PT } from 'prop-types';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { fetchAllClients } from '../../store/actions/clients';
+import { fetchAllTeams } from '../../store/actions/teams';
+import { fetchAllEmployees } from '../../store/actions/employees';
+import { getClientsLoading, getTeamsLoading, getEmployeesLoading } from '../../store/reducers';
+
+const AdminDashboardContainer = Wrapped =>
+  class extends React.Component {
+    static propTypes = {
+      clientsLoading: PT.object,
+      teamsLoading: PT.object,
+      employeesLoading: PT.object,
+      fetchAllClients: PT.func.isRequired,
+      fetchAllTeams: PT.func.isRequired,
+      fetchAllEmployees: PT.func.isRequired,
+    };
+
+    componentDidMount() {
+      this.getDashBoardInformation();
+    }
+    
+    getDashBoardInformation = async () => {
+      await this.props.fetchAllEmployees();
+      await this.props.fetchAllTeams();
+      await this.props.fetchAllClients();
```

Comment:
Do we need to await all of these calls?

Reply:
Yes, Errors were thrown when the three api calls were made at the same time.

Reply:
That sounds like a deeper issue to me; we should be able to dispatch multiple calls at the same time.

Reply:
That sounds like a deeper issue to me; we should be able to dispatch multiple calls at the same time.

Reply:
I'm not a fan of fetching all this information at the same time, especially saying some of it may never be seen.

For example the full list of employees will be a hefty response with a fair bit of data but if the /employees admin page is never viewed, that data is never seen.